client/src/main/java/io/pravega/client/tables/KeyValueTable.java

```diff
@@ -0,0 +1,258 @@
+ * Key Family; in this case the entire Key-Value Table load will be placed on a single backing Table Segment instead of
+ * spreading it across many Table Segments, leading to eventual performance degradation.
+ * </ul>
+ * <p>
+ * Types of Updates:
+ * <ul>
+ * <li> Unconditional Updates will insert and/or overwrite any existing values for the given Key, regardless of whether
+ * that Key previously existed or not, and regardless of what that Key's version is.
+ * <li> Conditional Updates will only overwrite an existing value if the specified version matches that Key's version.
+ * If the key does not exist, the {@link TableKey} or {@link TableEntry} must have been created with
+ * {@link KeyVersion#NOT_EXISTS} in order for the update to succeed.
+ * <li> Unconditional Removals will remove a Key regardless of what that Key's version is. The operation will also
+ * succeed (albeit with no effect) if the Key does not exist.
+ * <li> Conditional Removals will remove a Key only if the specified {@link TableKey#getVersion()} matches that Key's
+ * version. It will also fail (with no effect) if the Key does not exist and Version is not set to
+ * {@link KeyVersion#NOT_EXISTS}.
+ * </ul>
+ * <p>
+ * Conditional Update Responses:
+ * <ul>
+ * <li> Success: the update or removal has been atomically validated and performed; all updates or removals in the
+ * request have been accepted.
+ * <li> Failure: the update or removal has been rejected due to version mismatch; no update or removal has been performed.
+ * <li> {@link NoSuchKeyException}: the update or removal has been conditioned on a specific version (different from
+ * {@link KeyVersion#NOT_EXISTS} or {@link KeyVersion#NO_VERSION}) but the {@link TableKey} does not exist in the
+ * {@link KeyValueTable}.
+ * <li> {@link BadKeyVersionException}: the update or removal has been conditioned on a specific version (different from
+ * {@link KeyVersion#NO_VERSION} but the {@link TableKey} exists in the {@link KeyValueTable} with a different version.
+ * </ul>
+ *
+ * @param <KeyT>   Table Key Type.
+ * @param <ValueT> Table Value Type.
+ */
+public interface KeyValueTable<KeyT, ValueT> {
+    /**
+     * Unconditionally inserts a new or updates an existing Entry in the {@link KeyValueTable}.
+     *
+     * @param keyFamily (Optional) The Key Family for the Entry. If null, this Entry will not be associated with any
+     *                  Key Family.
+     * @param key       The Key to insert or update.
+     * @param value     The Value to be associated with the Key.
+     * @return A CompletableFuture that, when completed, will contain the {@link KeyVersion} associated with the newly
+     * inserted or updated entry.
+     */
+    CompletableFuture<KeyVersion> put(@Nullable String keyFamily, @NonNull KeyT key, @NonNull ValueT value);
+
+    /**
+     * Conditionally inserts a new Entry in the {@link KeyValueTable} if and only if the given Key is not already present.
```

Comment:
I suggest to rephrase this to the following:

```
Inserts a new Entry in the {@link KeyValueTable} only if the given Key is not already present.
```

The use of conditionally is redundant and the use of iff does not sound logically correct.

Reply:
removed `iff`